File: VB/T300213wf/App_Code/RepeaterTemplate.vb
Imports DevExpress.Web
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Web.UI
Imports System.Web.UI.HtmlControls
Imports System.Web.UI.WebControls

Public Class RepeaterTemplate
    Implements ITemplate

    Public Sub InstantiateIn(ByVal c As Control) Implements ITemplate.InstantiateIn
        Dim container As GridViewDataItemTemplateContainer = CType(c, GridViewDataItemTemplateContainer)
        Dim repeater As Repeater = New Repeater With {.ID = "repeater"}
        container.Controls.Add(New Literal With {.Text="<table class='innerTable'>"})
        container.Controls.Add(repeater)
        container.Controls.Add(New Literal With {.Text = "</table>"})

        repeater.DataSource = CType(container.Grid.GetRow(container.VisibleIndex), Product).Stores
        repeater.ItemTemplate = New RepeaterItemTemplate()
    End Sub
End Class

File: CS/T300213wf/App_Code/RepeaterTemplate.cs
using DevExpress.Web;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Web.UI;
using System.Web.UI.HtmlControls;
using System.Web.UI.WebControls;

public class RepeaterTemplate : ITemplate
{
	public void InstantiateIn(Control c) {
		GridViewDataItemTemplateContainer container = (GridViewDataItemTemplateContainer)c;
		Repeater repeater = new Repeater { ID = "repeater" };
		container.Controls.Add(new Literal { Text="<table class='innerTable'>"});
		container.Controls.Add(repeater);
		container.Controls.Add(new Literal { Text = "</table>" });

		repeater.DataSource = ((Product)container.Grid.GetRow(container.VisibleIndex)).Stores;
		repeater.ItemTemplate = new RepeaterItemTemplate();
	}
}
